src/test/test-execute.c

```diff
@@ -408,6 +408,47 @@ static void test_exec_inaccessiblepaths(Manager *m) {
         test(m, "exec-inaccessiblepaths-mount-propagation.service", can_unshare ? 0 : EXIT_FAILURE, CLD_EXITED);
 }
 
+static void test_exec_mount_apivfs(Manager *m) {
+        _cleanup_strv_free_ char **env = NULL;
+        const char *p;
+        int r;
+
+        FOREACH_STRING(p, "/usr/bin/touch", "/lib64/ld-linux-x86-64.so.2") {
+                r = find_executable(p, NULL);
+                if (r < 0) {
+                        log_notice_errno(r, "Skipping %s, could not find %s binary: %m", __func__, p);
+                        return;
+                }
+        }
+        if (find_executable("/lib64/libc.so.6", NULL) < 0) {
```

Comment:
This feels a bit too fragile - while unlikely to happen, the number of linked dependencies of a binary is not a stable api. We are building minimal images (chasing dependencies), could that be used to reproduce the same test instead? The /usr/share/minimal.raw stuff used in some other tests

Reply:
Yeah... But, could you suggest more in detail? I am not familiar with `RootImage/RootDirectory=`... In which test should I add the test? You mean the test should be added in TEST-50-DISSECT?

Reply:
It could still be part of the same test maybe? You wouldn't be able to use RootImage= because that implies MountAPIVFS, and if the point of the test is to disable that, it couldn't work.
So it would depend instead of minimal.raw, but rather than using it directly, it would unsquashfs and bind mount its /usr instead of the individual libraries. I think in principle this could work?